modules/blockexplorer/data.go

```diff
@@ -0,0 +1,66 @@
+package blockexplorer
+
+import (
+	"errors"
+
+	"github.com/NebulousLabs/Sia/build"
+	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/types"
+)
+
+// Calculates the total number of coins that have ever been created
+// from the bockheight
+func (be *BlockExplorer) totalCurrency() types.Currency {
+	totalCoins := uint64(0)
+	coinbase := types.InitialCoinbase
+	minCoinbase := types.MinimumCoinbase
+	for i := types.BlockHeight(0); i < be.blockchainHeight; i++ {
+		totalCoins += coinbase
+		if coinbase > minCoinbase {
+			coinbase--
+		}
+	}
+	return types.NewCurrency64(totalCoins).Mul(types.SiacoinPrecision)
+}
+
+// Returns a partial slice of our stored data on the blockchain. Data
+// obtained from consensus updates
+func (be *BlockExplorer) BlockInfo(start types.BlockHeight, finish types.BlockHeight) ([]modules.ExplorerBlockData, error) {
+	lockID := be.mu.RLock()
+	defer be.mu.RUnlock(lockID)
+
+	// Error checking on the given range
+	if start > finish {
+		return nil, errors.New("The start block must be higher than the end block")
```

Comment:
errors should have minimal punctuation, start with lower case letters and don't end with periods.

`the .. end block`